Lib9c/Action/ActionBase.cs

```diff
@@ -88,86 +96,94 @@ public AccountStateDelta(byte[] bytes)
             public IAccountStateDelta SetState(Address address, IValue state) =>
                 new AccountStateDelta(_states.SetItem(address, state), _balances);
 
-            public BigInteger GetBalance(Address address, Currency currency)
+            public FungibleAssetValue GetBalance(Address address, Currency currency)
             {
-                if (!_balances.TryGetValue((address, currency), out BigInteger balance))
+                if (!_balances.TryGetValue((address, currency), out (BigInteger, BigInteger) amount))
                 {
                     throw new BalanceDoesNotExistsException(address, currency);
                 }
 
-                return balance;
+                return new FungibleAssetValue(currency, amount.Item1, amount.Item2);
             }
 
-            public IAccountStateDelta MintAsset(Address recipient, Currency currency, BigInteger amount)
+            public IAccountStateDelta MintAsset(Address recipient, FungibleAssetValue value)
             {
                 // FIXME: 트랜잭션 서명자를 알아내 currency.AllowsToMint() 확인해서 CurrencyPermissionException
                 // 던지는 처리를 해야하는데 여기서 트랜잭션 서명자를 무슨 수로 가져올지 잘 모르겠음.
 
-                if (amount <= 0)
+                if (value <= new FungibleAssetValue(value.Currency))
                 {
-                    throw new ArgumentOutOfRangeException(nameof(amount));
+                    throw new ArgumentOutOfRangeException(nameof(value));
                 }
 
+                var nextAmount = GetBalance(recipient, value.Currency) + value;
+
                 return new AccountStateDelta(
                     _states,
-                    _balances.SetItem((recipient, currency),
-                        GetBalance(recipient, currency) + amount)
+                    _balances.SetItem(
+                        (recipient, value.Currency),
+                        (nextAmount.MajorUnit, nextAmount.MinorUnit)
+                    )
                 );
             }
 
             public IAccountStateDelta TransferAsset(
                 Address sender,
                 Address recipient,
-                Currency currency,
-                BigInteger amount,
+                FungibleAssetValue value,
                 bool allowNegativeBalance = false)
             {
-                if (amount <= 0)
+                if (value.Sign < 0)
```

Comment:
Is this the intended change?

Reply:
Yes, but `value.Sign < 0` isn't same as `amount <= 0` because it allows `0`. so I'll fix it.